pyccl/core.py

```diff
@@ -276,252 +296,219 @@ def __getitem__(self, key):
         except AttributeError:
             raise KeyError("Parameter '%s' not recognized." % key)
         return val
-    
+
     def __setitem__(self, key, val):
         """Set parameter values by name.
 
         """
         raise NotImplementedError("Parameters objects are immutable; create a "
                                   "new Parameters() instance instead.")
-        """
-        try:
-            # First check if the key already exists (otherwise the parameter 
-            # would be silently added to the ccl_parameters class instance)
-            getattr(self.parameters, key)
-        except AttributeError:
-            raise KeyError("Parameter '%s' not recognized." % key)
-        
-        # Set value of parameter
-        setattr(self.parameters, key, val)
-        # TODO: Should update/replace CCL objects appropriately
-        """
-    
+
     def __del__(self):
         """
-        Free the ccl_parameters instance that this Parameters object is 
+        Free the ccl_parameters instance that this Parameters object is
         managing.
         """
         if hasattr(self, 'parameters'):
-            if self.parameters is not None: 
+            if self.parameters is not None:
                 lib.parameters_free(self.parameters)
-    
-    
+
     def __str__(self):
         """Output the parameters that were set, and their values.
         """
         params = ['Omega_c', 'Omega_b', 'Omega_m', 'Omega_k', 'Omega_l',
-                  'w0', 'wa', 'H0', 'h', 'A_s', 'n_s', 'bcm_log10Mc', 
+                  'w0', 'wa', 'H0', 'h', 'A_s', 'n_s', 'bcm_log10Mc',
                   'bcm_etab', 'bcm_ks',
                   'Neff', 'mnu', 'Omega_n_mass', 'Omega_n_rel',
                   'T_CMB', 'Omega_g', 'z_star', 'has_mgrowth']
-        
+
         # Get values of parameters
         vals = []
         for p in params:
             try:
                 v = getattr(self.parameters, p)
-            except:
+            except Exception:
```

Comment:
What is the difference here between doing except and except Exception?

Reply:
Mike explained this in CR. Basically, this allows some exceptions to no be caught in some situations which is better. I forget the exact ones. This is def the right (tm) way to do things in python though.

Reply:
In particular, `KeyboardInterrupt`.  `except:` will catch this, which is almost certainly not what you want.  `except Exception:` does not, since `KeyboardInterrupt` derived directly from `BaseException`, not `Exception`.